### ISSUE TITLE
Nested block comments cause syntax error in PostgreSQL

### ISSUE THREAD

**Issue by nene** [sqlfluff/sqlfluff#5396]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the following code using `sqlfluff lint test.sql --dialect postgres`:

```sql
SELECT 1 /* hi hi /* foo */ ho ho */ AS bar;
```

### Expected Behaviour

Linting finishes without errors.

### Observed Behaviour

A parse error happens:

```
== [test.sql] FAIL                                                                                                            
L:   1 | P:  29 | AL02 | Implicit/explicit aliasing of columns.
                       | [aliasing.column]
L:   1 | P:  32 |  PRS | Line 1, Position 32: Found unparsable section: 'ho */ AS
                       | bar'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

The nested-comments syntax is part of [SQL standard](https://jakewheat.github.io/sql-overview/sql-2008-foundation-grammar.html#comment), though only a few dialects support it, among which to my knowledge are [PostgreSQL](https://www.postgresql.org/docs/8.0/sql-syntax.html#SQL-SYNTAX-COMMENTS), [DB2 for IBM i](https://www.ibm.com/docs/en/i/7.5?topic=statements-sql-comments) (a dialect not supported by SQLFluff) and [Transact-SQL](https://learn.microsoft.com/en-us/sql/t-sql/language-elements/slash-star-comment-transact-sql?view=sql-server-ver16).

Linting the code with `--dialect tsql` finishes without errors. So this feature is implemented for at least one dialect in SQLFluff. (Tried also all other supported dialects, they failed similarly to PostgreSQL.)

### How to reproduce

Simply try linting the code example above.

### Dialect

PostgreSQL

### Version

- SQLFluff 2.3.5
- Python 3.11.5

### Configuration

I don't have one.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-finley** [comment]
nb: we do have a db2 dialect: https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_db2.py

Thanks for raising this issue! Not sure how easy or hard it is to fix :)

**Comment by nene** [comment]
FYI, there are 3 dialects of DB2:

- DB2 for Unix, Linux, Windows (supported by SQLFluff)
- DB2 for IBM Z mainframe
- DB2 for IBM i

They all have slight differences in their syntax.
